core/templates/components/rubrics-editor/rubrics-editor.component.spec.ts

```diff
@@ -93,29 +93,40 @@ describe('Rubrics Editor Component', () => {
       .toBeTrue();
   });
 
-  it('should check for maximum character length of explanation', () => {
+  it('should check if explanations are at most 300 characters long', () => {
     let index: number = 2;
     componentInstance.ngOnInit();
-    componentInstance.editableExplanations[difficulty][index] = 'not_empty';
+    componentInstance.editableExplanations[difficulty][index] = 'a'.repeat(300);
     expect(componentInstance.isExplanationLengthValid(difficulty, index))
       .toBeTrue();
+    componentInstance.editableExplanations[difficulty][index] = 'a'.repeat(301);
+    expect(componentInstance.isExplanationLengthValid(difficulty, index))
+      .toBeFalse();
   });
 
   it('should check if medium level rubrics' +
       'have atleast one explantion',
   () => {
+    let index: number = 0;
+    expect(componentInstance.isMediumLevelExplanationValid()).toBeFalse;
     componentInstance.ngOnInit();
+    componentInstance.editableExplanations[difficulty][index] = 'not_empty';
     expect(componentInstance.isMediumLevelExplanationValid()).toBeTrue;
   });
 
-  it('should check check if total number of explanations' +
+  it('should check if total number of explanations' +
       'have reached the limit',
   () => {
-    let index: number = 2;
+    let index: number = 0;
     componentInstance.ngOnInit();
     componentInstance.editableExplanations[difficulty][index] = 'not_empty';
     expect(componentInstance.hasReachedExplanationCountLimit())
       .toBeFalse();
+    for (index = 1; index < 10; index++) {
+      componentInstance.editableExplanations[difficulty][index] = 'not_empty';
+    }
+    expect(componentInstance.hasReachedExplanationCountLimit())
+      .toBeTrue();
```

Comment:
Here the test case is failing, I'm not sure why is because as soon as the index will reach 10 there will be around 10 explanations and it should return true.
Thanks

Reply:
@EricZLou

Reply:
I'm guessing it may be an `IndexError`. Instead of indexing `...[difficulty][index] = 'not_empty'`, use `...[difficulty].push('not_empty')`

Reply:
Thank you so much, 
fixed.

Reply:
It's still failing at the same part wondering what's wrong, any idea @EricZLou?

Reply:
Does `hasReachedExplanationCountLimit` directly access `editableExplanations`? Try to add some print statements while using the dev server and see if something needs to be added or if we're pushing to the wrong list!

Reply:
I checked it and yes the values are getting reflected in `editableExplanations` as we go on with adding further explanations.
Thanks

Reply:
But I'm thinking that maybe pushing manually to `editableExplanations` doesn't make any updates to `this.rubrics[idx].getExplanations()` since that is what `hasReachedExplanationCountLimit` uses. Can you check if there's any difference? Maybe there's a better way to instantiate 10 explanations than pushing manually to `editableExplanations`.

Reply:
I can take a look at this later today if you'd like!

Reply:
Oh yeah, that might be possible, I'll take a look and see if I can find something related to it. Thank you for the info, it would be great if we'll be able to cross-check it.
Thanks

Reply:
Or maybe even `hasReachedExplanationCountLimit` should be accessing `editableExplanations` instead of `this.rubrics[idx].getExplanations()`? Feel free to explore this and see what makes sense.

Reply:
Point, that might be possible too. Thanks, I'll explore it and update you on the same.

Reply:
So, I checked it out and you were right that the `this.rubrics[idx].getExplanations()` is not getting updated when we add explanations to `editableExplanations` so I have updated the function and it's working fine.
Thanks for your support.

Reply:
Hey @EricZLou 
So this particular issue is resolved but I encountered another issue regarding the `accessing the property of undefined`, So in the recent commit I have changed the `hasReachedExplanationCountLimit` function code to check from the `editableExplanations` regarding the length of explanations and in the test file I was creating several explanations and adding them to `editableExplanations` to fail this test, Now as I'm getting this new error  I was thinking to maybe to revert back the  `hasReachedExplanationCountLimit` code and change the test file to add the explanations in that particular part but I don't think that would be a good idea right? I mean we will be making changes to the variable which is supposed to get updated by the parent entries?
Now I'm confused about how can I resolve this one.
Thanks